utils/build_swift/driver_arguments.py

```diff
@@ -713,99 +735,76 @@ def create_argument_parser():
-        title='Select the CMake generator')
-    cmake_generator_group.add_argument(
-        '-x', '--xcode',
-        action='store_const',
-        const='Xcode',
-        dest='cmake_generator',
-        help="use CMake's Xcode generator (default is Ninja)")
-    cmake_generator_group.add_argument(
-        '-m', '--make',
-        action='store_const',
-        const='Unix Makefiles',
-        dest='cmake_generator',
-        help="use CMake's Makefile generator (default is Ninja)")
-    cmake_generator_group.add_argument(
-        '-e', '--eclipse',
-        action='store_const',
-        const='Eclipse CDT4 - Ninja',
-        dest='cmake_generator',
-        help="use CMake's Eclipse generator (default is Ninja)")
-
+    # -------------------------------------------------------------------------
+    # Assertions group
+
+    with mutually_exclusive_group():
+        set_defaults(assertions=True)
+
+        # TODO: Convert to store_true
+        option('--assertions', store,
+               const=True,
+               help='enable assertions in all projects')
+
+        # TODO: Convert to store_false
+        option('--no-assertions', store('assertions'),
+               const=False,
+               help='disable assertions in all projects')
+
+    # -------------------------------------------------------------------------
+    in_group('Control assertions in a specific project')
+
+    option('--cmark-assertions', store,
+           const=True,
+           help='enable assertions in CommonMark')
+
+    option('--llvm-assertions', store,
+           const=True,
+           help='enable assertions in LLVM')
+    option('--no-llvm-assertions', store('llvm_assertions'),
+           const=False,
+           help='disable assertions in LLVM')
+
+    option('--swift-assertions', store,
+           const=True,
+           help='enable assertions in Swift')
+    option('--no-swift-assertions', store('swift_assertions'),
+           const=False,
+           help='disable assertions in Swift')
+
+    option('--swift-stdlib-assertions', store,
+           const=True,
+           help='enable assertions in the Swift standard library')
+    option('--no-swift-stdlib-assertions', store('swift_stdlib_assertions'),
+           const=False,
+           help='disable assertions in the Swift standard library')
+
+    option('--lldb-assertions', store,
+           const=True,
+           help='enable assertions in LLDB')
+    option('--no-lldb-assertions', store('lldb_assertions'),
+           const=False,
+           help='disable assertions in LLDB')
+
+    # -------------------------------------------------------------------------
+    in_group('Select the CMake generator')
+
+    set_defaults(cmake_generator=defaults.CMAKE_GENERATOR)
+
+    option(['-e', '--eclipse'], store('cmake_generator'),
+           const='Eclipse CDT4 - Ninja',
+           default=defaults.CMAKE_GENERATOR,
+           help="use CMake's Xcode generator (%(default)s by default)")
```

Comment:
This string seems accidentally repeated.

Overall very cool. 👏

Reply:
You are correct, thanks for catching that! I can't take all the credit here, this PR and the ones leading up to it were based on some awesome work by @rintaro last year in #2190.